shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPlugin.mm

```diff
@@ -0,0 +1,90 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPlugin.h"
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIApplication.h>
+#import <UIKit/UIKit.h>
+
+#include "flutter/fml/logging.h"
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterViewController_Internal.h"
+
+// Method Channel name to start spell check.
+static NSString* const kInitiateSpellCheck = @"SpellCheck.initiateSpellCheck";
+
+@interface FlutterSpellCheckPlugin ()
+
+@property(nonatomic, assign) FlutterMethodChannel* methodChannel;
+@property(nonatomic, retain) UITextChecker* textChecker;
+
+@end
+
+@implementation FlutterSpellCheckPlugin
+
+- (instancetype)initWithMethodChannel:(FlutterMethodChannel*)methodChannel {
+  self = [super init];
+  if (self) {
+    [self.methodChannel setMethodCallHandler:^(FlutterMethodCall* call, FlutterResult result) {
+      [self handleMethodCall:call result:result];
+    }];
+    self.textChecker = [[[UITextChecker alloc] init] retain];
+  }
+  return self;
+}
+
+- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
+  NSString* method = call.method;
+  id args = call.arguments;
+  if ([method isEqualToString:kInitiateSpellCheck]) {
+    result([self kInitiateSpellCheck:args]);
+  }
+}
+
+// Get the misspelled word and suggestions.
+//
+// The result will be formatted as a NSDictionary that looks like this:
+// {
+//   @"startIndex": 0,
+//   @"endIndex" : 5,
+//   @"suggestions": [@"suggestion1", @"suggestion2"..]
+// }
+- (NSDictionary*)kInitiateSpellCheck:(id)args {
+  NSString* language = args[0];
```

Comment:
Should this check if this is in `availableLanguages` and either bail if not, or allow nil and fall back to the first `availableLanguages` preference?  Doc says https://developer.apple.com/documentation/uikit/uitextchecker/1621033-availablelanguages?language=objc

> An array of strings representing ISO 639-1 language codes or combined ISO 639-1 language codes and ISO 3166-1 regional codes (for example, en_US).

Reply:
Actually, I can do some tests with an "unavailable" language and see what the return values are for rangeOfMisspelledWordInString. the result might just be as the same as bailing.

Reply:
The return value is (0, 0), so using a "unsupported" language is the same as "there is no spell checking error", which seems make sense to me.

The worry is that it makes harder for user to notice if they use a "unsupported" language. I think it is still necessary for us to check for it, and return an error in the method channel.

Reply:
Update:
No suggestions now returns an empty array
Unsupported language now returns nil.